app/backend/controllers/article.js

```diff
@@ -0,0 +1,188 @@
+/*
+  Post method for articles.
+  It saves article to database.
+*/
+module.exports.postArticle = async (request, response) => {
+    let Article = request.models['Article']
+  
+    // Article instance to add to the database
+    let article = new Article({
+      ...request.body,
+      userId: request.session['user']._id,
+      date: new Date()
+    });
+  
+    // Saves the instance into the database, returns any error occured
+    article.save()
+      .then(doc => {
+      return response.status(204).send();
+    }).catch(error => {
+      return response.status(400).send(error);
+    });
+  }
+  
+  /*
+    Get method for articles.
+    It returns given article.
+  */
+  module.exports.getArticle = async (request, response) => {
+    let Article = request.models['Article']
+
+    article = await Article.findOne({ _id : request.params['id']});
+  
+    if(article){
+      return response.send(article)
+    } else {
+      return response.status(400).send({
+        errmsg: "No such article."
+      })
+    }
+  }
+
+/*
+  Patch method for editing articles.
+  It saves article to database.
+*/
+module.exports.editArticle = async (request, response) => {
+  let Article = request.models['Article']
+  const userId = request.session['user']._id
+  const title = request.body["title"];
+  const text = request.body["text"];
+  const articleId = request.body['articleId'];
+
+  article = await Article.findOne({ _id : articleId});
+  if(article){
+    try{
+      if(!title && !text){ // no change
```

Comment:
you do not have to check whether title, text exist or not. client will request you in correct format, you only have to update the article with given id.

even if title did not change, client must send you existing title in the request body.

Reply:
you can think updating as semi-creating. the only difference is that you have to add new row to database with creating new _id. but in updating you change only existing row with given _id.